iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/listener/api/event/IcebergRequestContext.java

```diff
@@ -41,11 +42,24 @@ public class IcebergRequestContext {
    * @param catalogName The name of the catalog to be accessed in the request.
    */
   public IcebergRequestContext(HttpServletRequest httpRequest, String catalogName) {
+    this(httpRequest, catalogName, false);
+  }
+
+  /**
+   * Constructs a new {@code IcebergRequestContext} with specified HTTP request and catalog name.
+   *
+   * @param httpRequest The HttpServletRequest object containing request details.
+   * @param catalogName The name of the catalog to be accessed in the request.
+   * @param requestCredentialVending Whether the request is for credential vending.
+   */
+  public IcebergRequestContext(
+      HttpServletRequest httpRequest, String catalogName, boolean requestCredentialVending) {
     this.httpServletRequest = httpRequest;
     this.remoteHostName = httpRequest.getRemoteHost();
     this.httpHeaders = IcebergRestUtils.getHttpHeaders(httpRequest);
     this.catalogName = catalogName;
     this.userName = PrincipalUtils.getCurrentUserName();
+    this.requestCredentialVending = requestCredentialVending;
```

Comment:
Trying to get credential information from `httpHeaders`,  test jersey test fails for couldn't get real HTTP headers,  try find some solution